modulemd/modulemd-packager-v3.c

```diff
@@ -1628,3 +1628,146 @@ modulemd_packager_v3_parse_module_components (yaml_parser_t *parser,
 
   return TRUE;
 }
+
+
+gboolean
+modulemd_packager_v3_emit_yaml (ModulemdPackagerV3 *self,
+                                yaml_emitter_t *emitter,
+                                GError **error)
+{
+  MODULEMD_INIT_TRACE ();
+  g_autoptr (GError) nested_error = NULL;
+  gsize i;
+  g_autoptr (GPtrArray) keys = NULL;
+  gboolean ret;
+
+  /* Emit document headers */
+  if (!modulemd_yaml_emit_document_headers (
+        emitter, MODULEMD_YAML_DOC_PACKAGER, MD_PACKAGER_VERSION_THREE, error))
+    {
+      return FALSE;
+    }
+
+  /* Start data: */
+  EMIT_MAPPING_START (emitter, error);
+
+  /* TODO: emit data */
+  if (modulemd_packager_v3_get_module_name (self) != NULL)
+    {
+      EMIT_KEY_VALUE (
+        emitter, error, "name", modulemd_packager_v3_get_module_name (self));
+    }
+
+  if (modulemd_packager_v3_get_stream_name (self) != NULL)
+    {
+      EMIT_KEY_VALUE_FULL (emitter,
+                           error,
+                           "stream",
+                           modulemd_packager_v3_get_stream_name (self),
+                           YAML_DOUBLE_QUOTED_SCALAR_STYLE);
+    }
+
+  EMIT_KEY_VALUE (
+    emitter, error, "summary", modulemd_packager_v3_get_summary (self));
+  EMIT_KEY_VALUE_FULL (emitter,
+                       error,
+                       "description",
+                       modulemd_packager_v3_get_description (self),
+                       YAML_FOLDED_SCALAR_STYLE);
+
+  if (NON_EMPTY_TABLE (self->module_licenses))
+    {
+      EMIT_STRING_SET (emitter, error, "license", self->module_licenses);
+    }
+
+  if (self->xmd != NULL)
+    {
+      EMIT_SCALAR (emitter, error, "xmd");
+      if (!modulemd_yaml_emit_variant (emitter, self->xmd, error))
+        {
+          return FALSE;
+        }
+    }
+
+  /* TODO: fix this */
```

Comment:
This seems suspicious.

Reply:
This is also a leftover from my tailed attempt to use `EMIT_HASHTABLE_VALUES_IF_NON_EMPTY()`.